src/nix.rs

```diff
@@ -0,0 +1,338 @@
+                translate(rhs, file_id),
+            )
+            .into(),
+            More => Term::Op2(
+                BinaryOp::GreaterThan(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            LessOrEq => Term::Op2(
+                BinaryOp::LessOrEq(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            MoreOrEq => Term::Op2(
+                BinaryOp::GreaterOrEq(),
+                translate(lhs, file_id),
+                translate(rhs, file_id),
+            )
+            .into(),
+            NotEqual => Term::Op1(
+                UnaryOp::BoolNot(),
+                Term::Op2(
+                    BinaryOp::Eq(),
+                    translate(lhs, file_id),
+                    translate(rhs, file_id),
+                )
+                .into(),
+            )
+            .into(),
+
+            Implication => Term::App(
+                Term::App(
+                    Term::Op1(UnaryOp::Ite(), translate(lhs, file_id)).into(),
+                    translate(rhs, file_id),
+                )
+                .into(),
+                Term::Bool(true).into(),
+            )
+            .into(),
+
+            And => Term::App(
+                Term::Op1(UnaryOp::BoolAnd(), translate(lhs, file_id)).into(),
+                translate(rhs, file_id),
+            )
+            .into(),
+            Or => Term::App(
+                Term::Op1(UnaryOp::BoolOr(), translate(lhs, file_id)).into(),
+                translate(rhs, file_id),
+            )
+            .into(),
+        }
+    }
+}
+
+fn translate(node: rnix::SyntaxNode, file_id: FileId) -> RichTerm {
+    use rnix::types::{self, ParsedType, Wrapper};
+    use rnix::value;
+    use rnix::SyntaxKind::*;
+    use std::convert::{TryFrom, TryInto};
+    let pos = node.text_range();
+    let span = mk_span(file_id, pos.start().into(), pos.end().into());
+    println!("{:?}", node);
+    let node: ParsedType = node.try_into().unwrap();
+    match node {
+        ParsedType::Error(_) => Term::ParseError.into(),
+        ParsedType::Root(n) => translate(n.inner().unwrap(), file_id),
+        ParsedType::Paren(n) => translate(n.inner().unwrap(), file_id),
+        ParsedType::Dynamic(n) => translate(n.inner().unwrap(), file_id),
+
+        ParsedType::Assert(n) => unimplemented!(),
+
+        ParsedType::Value(n) => match n.to_value().unwrap() {
+            value::Value::Float(v) => Term::Num(v).into(),
+            value::Value::Integer(v) => Term::Num(v as f64).into(),
+            value::Value::String(v) => Term::Str(v).into(),
+            // TODO: How to manage Paths in nickel?
+            value::Value::Path(a, v) => Term::Str(v).into(),
+        },
+        ParsedType::Str(n) => {
+            let parts = n.parts();
+            //TODO: Do we actualy need the `Term::Str` variant in nickel AST?
```

Comment:
Yes, for evaluation. A `StrChunks` evaluate to a `Str` in the end, and operator like concatenation can only operate on strings. It's also useful to determine e.g. quoted fields that are actually statically known, as `{"myfield%bar" = 1}` from actualy dynamic fields, etc.

However, as far as compiling Nix is concerned, you can indeed stick to always producing `StrChunks`.

Reply:
Ok so a later optimisation could be done but we let it as it is for now.